src/main/java/TaskManager.java

```diff
@@ -0,0 +1,151 @@
+import java.util.Arrays;
+import java.util.ArrayList;
+
+public class TaskManager {
+    private TaskList tasks;
+
+    public TaskManager() {
+        this.tasks = new TaskList();
+    }
+
+    /**
+     * Parses the user input for keywords and commands.
+     * Calls the relevant methods to manage the list and print outputs to the user.
+     * @param userInput
+     * @return
+     */
+    public boolean parseInput(String userInput) {
+        String[] inputs = userInput.split(" ");
+
+        String keyword = inputs[0];
+
+        switch (keyword) {
+            case "bye":
+                return false;
+            case "list":
+                listTaskList();
+                return true;
+            
+        }
+        
+        if (inputs.length == 1) {
+            printError("You need to tell me more about what you want to do.");
+            return true;
+        }
+        
+        switch (keyword) {
+            case "done":
+                try {
+                    markComplete(Integer.parseInt(inputs[1]));
+                } catch (NumberFormatException e) {
+                    printError("You need to provide me with an integer index.");
+                }
+                return true;
+            case "delete":
+                try {
+                    deleteTask(Integer.parseInt(inputs[1]));
+                } catch (NumberFormatException e) {
+                    printError("You need to provide me with an integer index.");
+                }
+            case "todo":
+                addTodo(String.join(" ", Arrays.copyOfRange(inputs, 1, inputs.length)));
+                break;
+            case "deadline":
+                addDeadline(Arrays.copyOfRange(inputs, 1, inputs.length));
+                break;
+            case "event":
+                addEvent(Arrays.copyOfRange(inputs, 1, inputs.length));
+                break;
+            default: 
+                printError("Sorry. I don't know what that means");
+                return true;
+        }
+
+        System.out.println(String.format("  Now you have %s tasks in the list.", tasks.size()));
+
+        return true;
+    }
+    
+    private void deleteTask(int i) {
+
+    }
+
+    private void markComplete(int i) {
```

Comment:
it is better if alphabet 'i' is used only as a loop counter
consider using another name for the integer